config.c

```diff
@@ -302,6 +304,44 @@ void get_privacy_level(FILE *fp)
 			logg("Notice: Changing privacy level from %i to %i", config.privacylevel, value);
 			config.privacylevel = value;
 		}
+	}
+
+	// Release memory
+	release_config_memory();
+
+	// Have to close the config file if we opened it
+	if(opened)
+		fclose(fp);
+}
+
+void get_blocking_mode(FILE *fp)
+{
+	// Set default value
+	config.blockingmode = MODE_IP;
+
+	// See if we got a file handle, if not we have to open
+	// the config file ourselves
+	bool opened = false;
+	if(fp == NULL)
+	{
+		if((fp = fopen(FTLfiles.conf, "r")) == NULL)
+			// Return silently if there is no config file available
+			return;
+		opened = true;
+	}
```

Comment:
This seems messy. Why not just always open the file outside of the function and pass it in? Then it is not the function's responsibility to open the file (simpler function, easier to understand and debug).

Reply:
What if opening of the file fails? We would still have to check for if the file was successfully opened in the function to set at least the default value if it is not. Otherwise, if the function would only be called on a successful opening of the config file, it could not change the setting. The purpose here was not to keep the function most simple, but rather to keep the calling most simple (we just need to call `get_blocking_mode(NULL);` and it will take care of getting everything set up correctly). Note that we use a similar technique already in `get_privacy_level(NULL);` immediately above.